php/Terminus/Models/Environment.php

```diff
@@ -407,6 +407,63 @@ public function getConnectionMode() {
   }
 
   /**
+   * Retrieves recommended DNS settings for this environment
+   *
+   * @return array[]
+   * @throws TerminusException
+   */
+  public function getDns() {
+    if ($this->site->get('service_level') == 'free') {
+      throw new TerminusException(
+        'To add custom domains, this site must be upgraded to a paid plan.',
+        [],
+        1
+      );
+    }
+    $response = $this->request->simpleRequest(
+      sprintf(
+        'sites/%s/environments/%s?environment_state=true',
+        $this->site->get('id'),
+        $this->get('id')
+      )
+    );
+    $hosts          = $response['data']->urls;
+    $base_names     = $this->determineBaseNames($hosts);
+    $load_balancers = (array)$response['data']->loadbalancers;
+    $load_balancer  = array_shift($load_balancers);
+    $dns_settings   = [];
+    foreach ($hosts as $host) {
+      preg_match("/pantheon.io$/", $host, $matches);
+      if (empty($matches)) {
+        if (in_array($host, $base_names)) {
+          $dns_settings[] = [
+            'host'        => $host,
+            'record_type' => 'A',
+            'value'       => $load_balancer->ipv4,
+          ];
+          $dns_settings[] = [
+            'host'        => $host,
+            'record_type' => 'AAAA',
+            'value'       => $load_balancer->ipv6,
+          ];
+        } else {
+          $dns_settings[] = [
+            'host'        => $host,
+            'record_type' => 'CNAME',
+            'value'       => sprintf(
+              '%s-%s.pantheon.io',
+              $this->get('id'),
+              $this->site->get('name')
+            )
+          ];
+        }
+      }
+    }
```

Comment:
So let's see if I can get the correct logic right on the first try:

```
if an environment loadbalancer exists
  if it's a base domain 
    show the A record IP address
  else
     show CNAME to the pantheon basename
else if they are enterprise customer with custom LB (there is about 5 of these)
  if it's a base domain 
    show the A record IP address
  else
     show CNAME to the pantheon basename
else (no load balancer)
  if they have a gotpantheon.com domain
    show an A-record for legacy shared load balancer
  else
    if it's a base domain
      show the A record for shared www-redirector IP address and warn it's a redirector only
    else
       show the CNAME to the pantheon basename
```

And then ping @kibra to see if you got it correct. Or wait about 2 weeks and I'll see if I can cajole him into putting this into the API so we don't have to encode it in the client.